### ISSUE TITLE
Optimization to fuse getitem into read_parquet tasks

### ISSUE THREAD

**Issue by mrocklin** [dask/dask#1785]
### tl;dr

We should add a dask.dataframe optimization to fuse getitems on read_parquet tasks to produce new read_parquet tasks that only read the requested columns.

### Background

In #1782 we added basic support for Parquet, a fairly efficient and popular data format.  Parquet is a columnstore, and so it is much faster to pull out certain columns individually than to read through the entire file.  We can do this now explicitly when we read the parquet data.

    from dask.dataframe.io.parquet import read_parquet
    df = read_parquet('path/to/directory', columns=['a', 'b', 'c'])

However this requires foresight, and it is common to want to read a single column at a time from a larger dataframe

    df.a.sum().compute()

As written, this will still load all of df, then downselect.  We would rather have rewritten this as follows:

    read_parquet('path/to/directory', columns=['a']).a.sum().compute()

This is normally the sort of thing that a query optimizer does.  We don't have one of these.  However we *do* have task optimizers, and this simple replacement can be done in the common case.  In fact we already do this for bcolz and have a special function to help do it in that case, `fuse_getitem` (see `dask/dataframe/optimize.py`).

### Problems

So if we want to reuse the general `fuse_getitem` optimization we will need to simplify the currently generated graph a bit, which creates a few tasks for each partition, both because it uses from_delayed and because there is a separate set_index task.  This isn't hard, we would just need to create a new function that both reads from the row group and sets the index in one task and, instead of using from delayed we would construct the dictionary manually.

**Comment by mrocklin** [comment]
cc @jcrist @martindurant I suspect that either of you might enjoy the performance boost on dask.dataframe that comes from this task.
